torch_xla/experimental/fori_loop.py

```diff
@@ -21,22 +21,48 @@ def while_loop(cond_fn, body_fn, operands):
 
 def _xla_while_loop(cond_fn, body_fn, operands):
 
-  def op_fn(internal_x):
-    # TODO(manfei) replace cond_fn_placeholder and body_fn_placeholder once xla::while lowering in the backend is available
-    def cond_fn_placeholder(counter, internal_x):
-      return counter < xb.Op.scalar(internal_x.builder(), 10, dtype=xb.Type.S32)
-
-    def body_fn_placeholder(counter, internal_x):
-      next_counter = counter + xb.Op.scalar(
-          counter.builder(), 1, dtype=xb.Type.S32)
-      internal_x = internal_x + xb.Op.scalar(
-          internal_x.builder(), 1, dtype=xb.Type.S32)
-      return xb.Op.tuple((next_counter, internal_x))
-
-    zero = xb.Op.scalar(internal_x.builder(), 0, dtype=xb.Type.S32)
-    w = xb.Op.mkwhile((zero, internal_x), cond_fn_placeholder,
-                      body_fn_placeholder)
-    return w.get_tuple_element(1)
-
-  op = xor.register('test_while', op_fn)
-  return xu.as_list(op(operands[0]))
+  # create inputs placeholder
+  kwargs = {}
+  shapes = xb.tensor_shape(operands)
+  builder = xb.create_builder('test_while')
+  params = []
+  # secondparams = []
+  for shape in shapes:
+    p = xb.mkparam(builder, len(params), shape)
+    params.append(p)
+    # secondparams.append(xb.Op.tuple([p]))
+
+  # generate cond_fn xlacomputation
+  xm.mark_step()
```

Comment:
Let's keep this PR as draft until we experiment with `mark_step()`

Reply:
Thanks, we have a simple test PR for experimental of without `mark_step()`: https://github.com/pytorch/xla/pull/6643, will wait for test results in https://github.com/pytorch/xla/pull/6643